### PR TITLE
STR-196 consolidate provider store traits

### PR DESCRIPTION
**Pull request by sapinb**
## Description
* Merge *Provider and *Store traits to single *Database traits
* Adjust imports and variable names, etc as required 
* Rename all ChainState* -> Chainstate*

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [x] Refactor
-   [ ] New or updated tests

## Checklist

-   [x] I have performed a self-review of my code.
-   [ ] I have commented my code where necessary.
-   [x] I have updated the documentation if needed.
-   [x] My changes do not introduce new warnings.
-   [ ] I have added tests that prove my changes are effective or that my feature works.
-   [x] New and existing tests pass with my changes.

### CODE REVIEW

**Review by delbonis — CHANGES_REQUESTED** [review]
Broadly looks good.

File: crates/db/src/traits.rs [hunk 1]
```diff
@@ -170,7 +149,7 @@ pub trait ClientStateProvider {
 
 /// L2 data store for CL blocks.  Does not store anything about what we think
 /// the L2 chain tip is, that's controlled by the consensus state.
-pub trait L2DataStore {
+pub trait L2Database {
```
[the remark this answers is not in the document]
Comment (sapinb) [reply crates/db/src/traits.rs]: Updated

**Review by bewakes — APPROVED** [review]

**Review by delbonis — APPROVED** [review]
Seems good.

**Comment by delbonis** [comment]
Force merged because of codecov being annoying.

**Pull request merged**